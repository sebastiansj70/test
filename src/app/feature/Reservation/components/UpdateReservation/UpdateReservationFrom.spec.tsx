import '@testing-library/jest-dom';
import React from 'react';
import { RenderResult, render, wait, fireEvent } from '@testing-library/react';
import { UpdateReservation } from '.';
import { SinonStub, stub } from 'sinon';
import { setTextEvent } from 'app/shared/utils/test';

describe('UpdateReservationForm', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof UpdateReservation> & {
    onSubmit: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      onSubmit: stub(),
      formTitle: 'tittle form test',
      handleShow: jest.fn(),
      reservation: {
        idTicket: 1,
        telefonoUsuario: 3186905006,
        nombreUsuario: 'juan',
        horaIngreso: 1626283800000,
        horaSalida: 1626283800000,
        idCancha: 1,
        valor: 50000,
      },
    };
    componentWrapper = render(<UpdateReservation {...componentProps} />);
  });

  //   it('should match snapshot', () => {
  //     expect(componentWrapper.container).toMatchSnapshot();
  //   });

  it('debe fallar al faltar un dato', async () => {
    const elem = componentWrapper.container;
    const submitButton = elem.querySelector('button[type="submit"]');
    const valor = elem.querySelector('input[name="valor"]');

    await wait(() => {
      valor && fireEvent.change(valor, setTextEvent('valor', ''));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(1);
    expect(spans[0].textContent).toBe('El campo valor es requerido.');
  });

  it('deberia eniviar', async () => {
    const elem = componentWrapper.container;
    const telefonoUsuario = elem.querySelector('input[name="telefonoUsuario"]');
    const nombreUsuario = elem.querySelector('input[name="nombreUsuario"]');
    const horaIngreso = elem.querySelector('input[name="horaIngreso"]');
    const horaSalida = elem.querySelector('input[name="horaSalida"]');
    const idCancha = elem.querySelector('input[name="idCancha"]');
    const valor = elem.querySelector('input[name="valor"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      telefonoUsuario &&
        fireEvent.change(
          telefonoUsuario,
          setTextEvent('telefonoUsuario', '3186905006')
        );
    });

    await wait(() => {
      nombreUsuario &&
        fireEvent.change(nombreUsuario, setTextEvent('nombreUsuario', 'juan'));
    });

    await wait(() => {
      horaIngreso &&
        fireEvent.change(
          horaIngreso,
          setTextEvent('horaIngreso', '1626283800000')
        );
    });

    await wait(() => {
      horaSalida &&
        fireEvent.change(
          horaSalida,
          setTextEvent('horaSalida', '1626283800000')
        );
    });

    await wait(() => {
      idCancha && fireEvent.change(idCancha, setTextEvent('idCancha', '1'));
    });

    await wait(() => {
      valor && fireEvent.change(valor, setTextEvent('valor', '50000'));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });

    const formSubmitted = componentProps.onSubmit.firstCall.args[1];

    expect(formSubmitted.telefonoUsuario).toBe(3186905006);
    expect(formSubmitted.nombreUsuario).toBe('juan');
    expect(formSubmitted.horaIngreso).toBe(1626283800000);
    expect(formSubmitted.horaSalida).toBe(1626283800000);
    expect(formSubmitted.idCancha).toBe(1);
    expect(formSubmitted.valor).toBe(50000);
  });
});
