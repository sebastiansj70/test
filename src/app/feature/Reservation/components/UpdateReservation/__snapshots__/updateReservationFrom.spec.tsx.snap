// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateReservationForm should match snapshot 1`] = `
<div>
  <form
    class="sc-AxiKw iPybUT"
  >
    <h1
      class="sc-AxgMl jfqWoR"
    >
      tittle form test
    </h1>
    <input
      class="sc-AxhUy cAHivH"
      name="telefonoUsuario"
      placeholder=""
      value="3186905006"
    />
    <input
      class="sc-AxhUy cAHivH"
      name="nombreUsuario"
      placeholder=""
      value="juan"
    />
    <input
      class="sc-AxhUy cAHivH"
      name="horaIngreso"
      placeholder=""
      value="1626283800000"
    />
    <input
      class="sc-AxhUy cAHivH"
      name="horaSalida"
      placeholder=""
      value="1626283800000"
    />
    <input
      class="sc-AxhUy cAHivH"
      name="idCancha"
      placeholder=""
      value="1"
    />
    <input
      class="sc-AxhUy cAHivH"
      name="valor"
      placeholder=""
      value="50000"
    />
    <button
      class="sc-AxjAm kgxTKy"
      name="Actualizar"
      type="submit"
    >
      Actualizar
    </button>
  </form>
</div>
`;
